lib/iris/tests/test_netcdf.py

```diff
@@ -455,69 +490,77 @@ def test_lat_lon_earth_radius(self):
         # so that it has the same signature as other coord builder routines.
         engine = None
         crs = ncload_helpers.build_coordinate_system(engine, self.grid)
-        self.assertEqual(crs, icoord_systems.GeogCS(earth_radius))
+        assert crs == icoord_systems.GeogCS(earth_radius)
 
 
-class SaverPermissions(tests.IrisTest):
-    def test_noexist_directory(self):
+class TestSaverPermissions:
+    def test_noexist_directory(self, tmp_path):
         # Test capture of suitable exception raised on writing to a
         # non-existent directory.
-        dir_name = os.path.join(tempfile.gettempdir(), "non_existent_dir")
-        fnme = os.path.join(dir_name, "tmp.nc")
-        with self.assertRaises(IOError):
+        fnme = str(tmp_path / "non_existent_dir" / "tmp.nc")
+        with pytest.raises(IOError, match="Permission denied"):
             with iris.fileformats.netcdf.Saver(fnme, "NETCDF4"):
                 pass
 
-    def test_bad_permissions(self):
+    @pytest.fixture
+    def tmp_path_no_write(self, tmp_path):
+        dir = tmp_path / "no_write_dir"
+        dir.mkdir()
+        original_perms = dir.stat().st_mode
+        dir.chmod(stat.S_IREAD | stat.S_IEXEC)
+        yield dir
+        dir.chmod(original_perms)
```

Comment:
No criticism, I just like this fixture. If I could think of more uses for it I'd say put it in conftest, just cos it's purdy.